src/shogun/machine/MulticlassMachine.cpp

```diff
@@ -51,7 +51,7 @@ CMulticlassMachine::~CMulticlassMachine()
 
 void CMulticlassMachine::set_labels(CLabels* lab)
 {
-    CMachine::set_labels(lab);
+    CMachine::set_labels(dynamic_cast<CMulticlassLabels*>(lab));
```

Comment:
I have run `shogun-unit-test` before I sent this pr, I am sorry I dont know where I am wrong, you mean I need create a small test or example passing `CBinaryLabels ` to `CMulticlassClassifier`?